src/nvim/hardcopy.c

```diff
@@ -455,8 +454,10 @@ static void prt_line_number(prt_settings_T *psettings, int page_line, linenr_T l
  */
 int prt_header_height(void)
 {
-  if (printer_opts[OPT_PRINT_HEADERHEIGHT].present)
-    return printer_opts[OPT_PRINT_HEADERHEIGHT].number;
+  if (printer_opts[OPT_PRINT_HEADERHEIGHT].present) {
+    assert(printer_opts[OPT_PRINT_HEADERHEIGHT].number <= INT_MAX);
```

Comment:
Ok, I reviewed every place where INT_MAX was introduced to check conversion to int, and added the corresponding check for INT_MIN when converting from a signed type. Probably, there are cases where, while the variable is signed, it is meant to hold a positive value (mouse_col, mouse_row, etc). In those cases, asserting >= 0 && <= INT_MAX would be more sensible. But, without deeper understanding of the code, and not knowing if value can ever be negative for some reason, I prefer the safe path of asserting >= INT_MIN && <= INT_MAX.

Note also that a lot more of type refactoring cleanup can (and should) still be done in these files, which would eventually eliminate most asserts/casts. But that's not this PR's scope. Here I'm just doing the minimum to be able to enable -Wconversion and remove long_u.

Pushed new version with these changes.